### PR TITLE
use inClusterConfig before using default config for cluster components

### PR DESCRIPTION
**Pull request by mikedanese**

### CODE REVIEW

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -415,6 +407,26 @@ func containsVersion(versions *unversioned.APIVersions, version string) bool {
 	return false
 }
 
+func (s *CMServer) buildKubeconfig() (*client.Config, error) {
```
Comment (roberthbailey) [inline cmd/kube-controller-manager/app/controllermanager.go]: This function is identical in all three files. Can we abstract it out (by passing in the kubeconfig & master args) and just implement it once? Then next time we want to tweak the behavior we won't have to do it in three places.

Comment (roberthbailey) [inline cmd/kube-controller-manager/app/controllermanager.go]: ping...

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by mikedanese** [comment]
@roberthbailey fixed. PTAL

**Comment by roberthbailey** [comment]
LGTM.

**Pull request merged**